internal/transport/http_util_test.go

```diff
@@ -22,28 +22,46 @@ import (
 	"errors"
 	"fmt"
 	"io"
+	"math"
 	"net"
 	"reflect"
 	"testing"
 	"time"
 )
 
-func (s) TestTimeoutDecode(t *testing.T) {
+func (s) TestDecodeTimeout(t *testing.T) {
 	for _, test := range []struct {
 		// input
 		s string
 		// output
 		d   time.Duration
 		err error
 	}{
+		{"0S", 0, nil},
```

Comment:
The [spec](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#requests) mentions that `TimeoutValue` should be a positive integer. So this should ideally return an error on validation.

Reply:
Done.